src/main/java/edu/kit/datamanager/metastore2/web/impl/SchemaRegistryControllerUIImpl.java

```diff
@@ -56,59 +52,80 @@
+    @ResponseBody
+    public ResponseEntity<TabulatorRemotePagination> getRecordsForUi(Pageable pgbl, WebRequest wr, HttpServletResponse hsr, UriComponentsBuilder ucb){
 
-    @RequestMapping("/schema-management")
-    @Override
-    public ModelAndView schemaManagement() {
-        // Search for resource type of MetadataSchemaRecord
-        Specification<DataResource> spec = ResourceTypeSpec.toSpecification(ResourceType.createResourceType(MetadataSchemaRecord.RESOURCE_TYPE));
+       Pageable pageable = PageRequest.of(pgbl.getPageNumber()-1, pgbl.getPageSize(), Sort.by("id").ascending());
+         
+       ResponseEntity<List<MetadataSchemaRecord>>  responseEntity4schemaRecords = schemaControllerImpl.getRecords(null, null, null, null, pageable,  wr,  hsr,  ucb);      
+        List<MetadataSchemaRecord> schemaRecords = responseEntity4schemaRecords.getBody();
+        
+        String pageSize =  responseEntity4schemaRecords.getHeaders().getFirst("Content-Range");
+        
+        TabulatorRemotePagination tabulatorRemotePagination = TabulatorRemotePagination.builder()
+                .lastPage((Integer.parseInt(pageSize.split("/")[1])/pageable.getPageSize())+1)
+                .data(schemaRecords)
+                .build();
+        
+        return ResponseEntity.status(HttpStatus.OK).body(tabulatorRemotePagination);
 
-        List<DataResource> records = null;
-        records = schemaRecordDao.findAll(spec);
+    }
+    
+    @RequestMapping("/api/v1/ui/metadataBySchemaId/{id}")
+    @ResponseBody
+    public ResponseEntity<TabulatorRemotePagination> getMetadataRecordsForUi(@PathVariable(value = "id", required = true) String id, Pageable pgbl, WebRequest wr, HttpServletResponse hsr, UriComponentsBuilder ucb){
+
+        Pageable pageable = PageRequest.of(pgbl.getPageNumber()-1, pgbl.getPageSize(), Sort.by("id").ascending());
+
+        ResponseEntity< List<MetadataRecord>> responseEntity4metadataRecords = metadtaControllerImpl.getRecords(null, null, Arrays.asList(id),null,null, pageable, wr, hsr, ucb);
+        List<MetadataRecord> metadataRecords = responseEntity4metadataRecords.getBody();
+
+        String pageSize =  responseEntity4metadataRecords.getHeaders().getFirst("Content-Range");
+        
+        TabulatorRemotePagination tabulatorRemotePagination = TabulatorRemotePagination.builder()
+                .lastPage((Integer.parseInt(pageSize.split("/")[1])/pageable.getPageSize())+1)
+                .data(metadataRecords)
+                .build();
+        
+        return ResponseEntity.status(HttpStatus.OK).body(tabulatorRemotePagination);
 
-        List<MetadataSchemaRecord> schemaRecords = new ArrayList<>();
-        records.forEach((record) -> {
-            MetadataSchemaRecord item = MetadataSchemaRecordUtil.migrateToMetadataSchemaRecord(schemaConfig, record, false);
-            fixSchemaDocumentUri(item);
-            schemaRecords.add(item);
-        });
+    }
 
+    @RequestMapping("/schema-management")
```

Comment:
All endpoints should contain the version number to avoid later conflicts. ( "/schema-management" -> "/api/v1/schema-management") 
Alternatively: /api/v1/ui/schema-management"